src/puppetlabs/trapperkeeper/services_internal.clj

```diff
@@ -87,64 +87,36 @@
             optional-form (array-map dep optional-dep-default)]
         (recur (rest optional) (conj output optional-form))))))
 
-;; TODO Not converting to use schema since this fn's behavior should literally
-;; be a schema
-(defn validate-fn-forms!
-  "Validate that all of the fn forms in the service body appear to be
-  valid fn definitions.  Throws `IllegalArgumentException` otherwise."
-  [fns]
-  {:pre [(seq? fns)]
-   :post [(map? %)
-          (= #{:fns} (set (keys %)))]}
-  (if (every? seq? fns)
-    {:fns fns}
-    (throw (IllegalArgumentException.
-            (format
-             "Invalid service definition; expected function definitions following dependency list, invalid value: '%s'"
-             (pr-str (first (filter #(not (seq? %)) fns))))))))
-
-;; TODO Not converting to use schema since this fn's behavior should literally
-;; be a schema
-(defn validate-deps-form!
-  "Validate that the service body has a valid dependency specification.
-  Throws `IllegalArgumentException` otherwise."
-  [forms]
-  {:pre [(seq? forms)]
-   :post [(map? %)
-          (= #{:fns :dependencies} (set (keys %)))]}
-  (let [f (first forms)
-        deps (cond
-               (vector? f) f
-               (map? f) (transform-deps-map f)
-               :else (throw (IllegalArgumentException.
-                             (format
-                              "Invalid service definition; expected dependency list following protocol, found: '%s'"
-                              (pr-str f)))))]
-    (merge {:dependencies deps} (validate-fn-forms! (rest forms)))))
-
-;; TODO Not converting to use schema since this fn's behavior should literally
-;; be a schema
-(defn find-prot-and-deps-forms!
+(schema/defn ^:always-validate find-prot-and-deps-forms!
+  :- {:fns (schema/pred seq?) :dependencies (schema/pred vector?) :service-protocol-sym (schema/maybe Symbol)}
```

Comment:
It's probably worth extracting this schema to be a named schema at the top of the file and then referring to the name here.  Maybe call it ProtocolAndDependencies or ProtocolAndDependenciesForms or something.